syntax = "proto3";

package schemas;


message BFloat16Tensor {
    bytes data = 1;  // 使用 bytes 存储 bfloat16 数据
    repeated int32 shape = 2;  // 形状信息
}

message InputIds {
    repeated int32 input_ids = 1;
}

message ForwardRequest {
    repeated string uuid_list = 1;
    repeated InputIds input_ids_list = 2;
    BFloat16Tensor hidden_states = 3;
}

message StatusRequest {
    repeated string uuid = 1;
    int32 pp_idx = 2;
    float cost_time = 3;
}

message StatusResponse {
    string msg = 1;
    int32 status = 2;
}

message ForwardResponse {
    string msg = 1;
    int32 status = 2;
}

message HealthResponse {
    string msg = 1;
    int32 status = 2;
}

message Empty {}

message SetConfigRequest {
    string forward_url = 1;
    string master_url = 2;
    int32 pp_rank = 3;
}

message SetConfigResponse {
    string msg = 1;
    int32 status = 2;
}

message ImageForwardRequest {
    repeated string uuid = 1;
    BFloat16Tensor hidden_states = 2;
    BFloat16Tensor encoder_hidden_states = 3;
    BFloat16Tensor text_embeddings = 4;
    int32 seq_len = 5;
    int32 height = 6;
    int32 width = 7;
}



service RPCService {
    rpc Status (StatusRequest) returns (StatusResponse);
    rpc Forward (ForwardRequest) returns (ForwardResponse);
    rpc Health (Empty) returns (HealthResponse);
    rpc SetConfig (SetConfigRequest) returns (SetConfigResponse);
    rpc ImageForward (ImageForwardRequest) returns (ForwardResponse);
}
